integration/session_test.rb

```diff
@@ -772,4 +772,52 @@ def test_host_broadcast_listen_address
     cluster && cluster.close
   end
 
+  # Test for setting the protocol_version
+  #
+  # test_can_set_protocol_version_explicitly tests that the protocol_version for a cluster object can be set, such that
+  # this version is used explicitly when connecting to the Cassandra cluster, rather than automatically negotiating it.
+  # It first tests that a valid protocol version can be set. It then tests that setting a protocol version lower than
+  # the Cassandra cluster supports throws a Cassandra::Errors::NoHostsAvailable. Finally, it tests that setting a protocol
+  # version higher than the Cassandra cluster supports throws a Cassandra::Errors::ProtocolError.
+  #
+  # @expected_errors [Cassandra::Errors::NoHostsAvailable] When a protocol version lower than supported is specified.
+  # @expected_errors [Cassandra::Errors::ProtocolError] When a protocol version higher than supported is specified.
+  #
+  # @since 3.0.0
+  # @jira_ticket RUBY-165
+  # @expected_result protocol versions specified should be honored when connecting to the Cassandra cluster.
+  #
+  # @test_category connection
+  #
+  def test_can_set_protocol_version_explicitly
+    ## Setting a valid protocol version
+    if CCM.cassandra_version < '3.0.0'
+      cluster = Cassandra.cluster(protocol_version: 1)
+      assert_equal 1, cluster.instance_variable_get(:@connection_options).protocol_version
+    else
+      cluster = Cassandra.cluster(protocol_version: 3)
+      assert_equal 3,  cluster.instance_variable_get(:@connection_options).protocol_version
+    end
+
+    ## Setting protocol version lower than supported
+    # Protocol version 0 is invalid
+    assert_raises(Cassandra::Errors::NoHostsAvailable) do
+      Cassandra.cluster(protocol_version: 0)
```

Comment:
This will likely fail a different way after you get my validation update. I just merged my fix to master, so merge down to your branch and then see how this test needs to be updated.